frontend/src/contexts/AuthContext.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { createContext, useEffect, useState } from "react";
+
+type AuthState = {
+  isLoggedIn: boolean;
+  userEmail: string | undefined;
+  login: (username: string, password: string) => Promise<boolean> | undefined;
+  verifyLogin: () => Promise<boolean> | undefined;
+  logout: () => Promise<boolean> | undefined;
+};
+
+const initialState: AuthState = {
+  isLoggedIn: false,
+  userEmail: undefined,
+  login: () => undefined,
+  verifyLogin: () => undefined,
+  logout: () => undefined,
+};
+
+const SERVER_URL = "http://localhost:3001";
```

Comment:
Like in the backend this should a .env file

An example of how it would look like in the frontend is [here](https://github.com/TritonSE/SDCTA-Data-Visualization-Website/blob/main/SDCTA/src/api/consumer.ts#L4C13-L4C54)

Also the dependence should be `@types/node`